src/mempool_limit.h

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) 2019 The Zcash developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or https://www.opensource.org/licenses/mit-license.php .
+
+#ifndef MEMPOOLLIMIT_H
+#define MEMPOOLLIMIT_H
+
+#include <map>
+#include <set>
+#include <vector>
+
+#include "boost/optional.hpp"
+#include "primitives/transaction.h"
+#include "uint256.h"
+
+const size_t DEFAULT_MEMPOOL_TOTAL_COST_LIMIT = 80000000;
+const int64_t DEFAULT_MEMPOOL_EVICTION_MEMORY_MINUTES = 60;
+
+const size_t RECENTLY_EVICTED_SIZE = 40000;
+const uint64_t MIN_TX_WEIGHT = 4000;
+const uint64_t LOW_FEE_PENALTY = 16000; 
+
+class RecentlyEvictedList
+{
+    const size_t capacity;
+
+    const int64_t timeToKeep;
+    // Pairs of txid and time (seconds since epoch)
+    std::deque<std::pair<uint256, int64_t>> txIdsAndTimes;
+    std::set<uint256> txIdSet;
```

Comment:
I figured out how to do an unordered set (more efficient), if you'd like. The gtests pass with this change.
```
diff --git a/src/mempool_limit.h b/src/mempool_limit.h
index 63b24eaa1..89cf405c6 100644
--- a/src/mempool_limit.h
+++ b/src/mempool_limit.h
@@ -6,7 +6,7 @@
 #define MEMPOOLLIMIT_H
 
 #include <map>
-#include <set>
+#include <unordered_set>
 #include <vector>
 
 #include "boost/optional.hpp"
@@ -20,6 +20,13 @@ const size_t RECENTLY_EVICTED_SIZE = 40000;
 const uint64_t MIN_TX_WEIGHT = 4000;
 const uint64_t LOW_FEE_PENALTY = 16000; 
 
+class RecentlyEvictedListHasher {
+public:
+    size_t operator()(const uint256& key) const {
+        return key.GetCheapHash();
+    }
+};
+
 class RecentlyEvictedList
 {
     const size_t capacity;
@@ -27,7 +34,7 @@ class RecentlyEvictedList
     const int64_t timeToKeep;
     // Pairs of txid and time (seconds since epoch)
     std::deque<std::pair<uint256, int64_t>> txIdsAndTimes;
-    std::set<uint256> txIdSet;
+    std::unordered_set<uint256, RecentlyEvictedListHasher> txIdSet;
 
     void pruneList();
 
```

Reply:
In case it's not obvious, this suggestion is non-blocking. We can possibly do this later, or maybe create a tech-debt PR that replaces all unnecessarily-ordered sets and maps with unordered ones.

Reply:
Nice. Looks great! At this point I think I would be happier fixing these as part of a follow up, but if other reviewers think this should this should be included as part of this PR I am happy to do that to.